operator/src/test/java/org/bf2/operator/managers/IngressControllerManagerTest.java

```diff
@@ -552,6 +555,29 @@ void testIngressControllerInternal() {
         assertEquals("Internal", controller.getSpec().getEndpointPublishingStrategy().getLoadBalancer().getScope());
     }
 
+    @Test
+    void testIngressControllerGcp() {
+        KafkaInstanceConfigurations mock = Mockito.mock(KafkaInstanceConfigurations.class);
+        Mockito.when(mock.getPlatform()).thenReturn(Platform.GCP);
+        QuarkusMock.installMockForType(mock, KafkaInstanceConfigurations.class);
+
+        ManagedKafkaAgent agent = new ManagedKafkaAgentBuilder()
+                .withNewMetadata()
+                .withName(ManagedKafkaAgentResourceClient.RESOURCE_NAME)
+                .endMetadata()
+                .withNewSpec()
+                .editOrNewNet().withPrivate(true).endNet()
+                .endSpec()
+                .build();
+        IngressController controller = ingressControllerManager.buildIngressController("kas", "kas", null, 1, null, null, agent);
+        assertEquals("---\n"
+                + "aws:\n"
+                + "  type: \"NLB\"\n"
+                + "gcp:\n"
+                + "  clientAccess: \"Global\"\n"
```

Comment:
Presumably having entries for providers other than the one the cluster is actually using is harmless. I suppose the experience with having AWS all this time on GCP proves that.

Reply:
Yes it's an odd system.  The type has to match before the cluster ingress operator will even look at the parameters, so the aws params have been ignored this whole time on gcp.  Also the operator won't even look at the clientAccess unless the scope is internal.